streams/runners/storm/actions/src/main/java/com/hortonworks/streamline/streams/actions/storm/topology/ServiceConfigurationReader.java

```diff
@@ -0,0 +1,99 @@
+package com.hortonworks.streamline.streams.actions.storm.topology;
+
+import com.google.common.collect.Lists;
+import com.hortonworks.streamline.common.QueryParam;
+import com.hortonworks.streamline.streams.catalog.Cluster;
+import com.hortonworks.streamline.streams.catalog.Namespace;
+import com.hortonworks.streamline.streams.catalog.NamespaceServiceClusterMapping;
+import com.hortonworks.streamline.streams.catalog.ServiceConfiguration;
+import com.hortonworks.streamline.streams.cluster.service.EnvironmentService;
+import com.hortonworks.streamline.streams.storm.common.ServiceConfigurationReadable;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class ServiceConfigurationReader implements ServiceConfigurationReadable {
+
+    private final EnvironmentService environmentService;
+    private final long namespaceId;
+
+    public ServiceConfigurationReader(EnvironmentService environmentService, long namespaceId) {
+        this.environmentService = environmentService;
+        this.namespaceId = namespaceId;
+    }
+
+    @Override
+    public Map<Long, Map<String, String>> readAllClusters(String serviceName) {
+        Namespace namespace = environmentService.getNamespace(namespaceId);
+
+        if (namespace == null) {
+            throw new IllegalArgumentException("Namespace " + namespaceId + " doesn't exist.");
+        }
+
+        Long namespaceId = namespace.getId();
+
+        Collection<NamespaceServiceClusterMapping> mappings = environmentService.listServiceClusterMapping(namespaceId, serviceName);
+
+        List<Long> clusters = mappings.stream()
+                .map(mapping -> mapping.getClusterId())
+                .collect(Collectors.toList());
+
+        Map<Long, Map<String, String>> retMap = new HashMap<>();
+        clusters.forEach(c -> {
+            Map<String, String> flattenConfig = read(c, serviceName);
+            retMap.put(c, flattenConfig);
+        });
+
+        return retMap;
+    }
+
+    @Override
+    public Map<String, String> read(Long clusterId, String serviceName) {
+        Collection<NamespaceServiceClusterMapping> mappings = environmentService.listServiceClusterMapping(namespaceId, serviceName);
```

Comment:
Here we get namespace-service mapping, and see whether given cluster is associated for the service. If it isn't, don't retrieve configurations.